src/nvim/terminal.c

```diff
@@ -2029,6 +2039,12 @@ static void refresh_terminal(Terminal *term)
 
   int ml_added = buf->b_ml.ml_line_count - ml_before;
   adjust_topline(term, buf, ml_added);
+
+  // Copy pending events back to the main event queue
+  while (!multiqueue_empty(term->pending.events)) {
+    Event event = multiqueue_get(term->pending.events);
+    multiqueue_put_event(main_loop.events, event);
```

Comment:
This pattern might be useful in more places, WDYT about adding `multiqueue_move_events` ?

Reply:
Yea, we use it in at least one other place already. I'll add that.